Lab3.ipynb

```diff
@@ -0,0 +1,405 @@
+    "badData = np.asarray(badData)\n",
+    "badKept = badData[np.where((badData>12-sigma*.4) & (badData<12+sigma*.4))]\n",
+    "badDiscard = len(badData) - len(badKept)\n",
+    "print('Sigma value:', sigma)\n",
+    "print('Pos True:', len(goodKept))\n",
+    "print('Pos False:', len(badKept))\n",
+    "print('Neg True:', badDiscard)\n",
+    "print('Neg False:', goodDiscard)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 1.A.4: According to my truth table for a 5 sigma stastistical significance probability, we are keeping 2 bad data values and all the good data values. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 117,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sigma value: 1\n",
+      "Pos True: 68171\n",
+      "Pos False: 0\n",
+      "Neg True: 8\n",
+      "Neg False: 31829\n",
+      "\n",
+      "Sigma value: 2\n",
+      "Pos True: 95451\n",
+      "Pos False: 1\n",
+      "Neg True: 7\n",
+      "Neg False: 4549\n",
+      "\n",
+      "Sigma value: 3\n",
+      "Pos True: 99717\n",
+      "Pos False: 1\n",
+      "Neg True: 7\n",
+      "Neg False: 283\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# 1.B:\n",
+    "for i in range(1,4): \n",
+    "    goodKept = goodData[np.where((goodData< 12+i*.4) & (goodData> 12-i*.4))]\n",
+    "    goodDiscard = len(goodData) - len(goodKept)\n",
+    "    badData = np.asarray(badData)\n",
+    "    badKept = badData[np.where((badData>12-i*.4) & (badData<12+i*.4))]\n",
+    "    badDiscard = len(badData) - len(badKept)\n",
+    "    print('Sigma value:', i)\n",
+    "    print('Pos True:', len(goodKept))\n",
+    "    print('Pos False:', len(badKept))\n",
+    "    print('Neg True:', badDiscard)\n",
+    "    print('Neg False:', goodDiscard)\n",
+    "    print()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 1.B: Above I reran the code with different sigma values and as you can see the larger the threshhold the less likely you are to throw away good data. This is because more data will meet the demands of our threshold as the tolerancesa relax. And these omissions are predictable since we are dealing with a gaussian distribution. We know that within 1, 2, and 3 standard deviations the percentage of data inside that range is 68%, 95%, and 99.7% respectively."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Problem 1.C: Yes, there can be mistakes of commission. In our example with 5 sigma we had two bad data points. However, they are avoidable by making the statistical threshold smaller. However, this comes at a price described in the answer above. If we decrease the threshold too much then we omit more good data. So finding a good balance between accepting good data and rejecting bad data will yield the most optimal threshold. "
```

Comment:
Problem 1.C:
- explanation is good and makes sense; consider finding the "good balance" that you discussed too (in mine, I guessed a sigma slightly above 4.1 to minimize both commission and omission mistakes, while prioritizing keeping commission low, since we *really* don't want bad data getting in)